### PR TITLE
MWPW-157346:[NALA] Move Nala tests to Milo repository

### PR DESCRIPTION
**Pull request by skumar09**
Move Nala UI tests to the Milo repository.

- Define Nala test structures, format, and organization in the Milo repository.
- Define how to create, run, and debug Nala UI tests from the Milo repository.
- Provide options to run Nala UI tests on 
      - Local env
      - Libs env
      - Milolibs 
      - Stage, Main, feature branch, or any environments
 - Provide options to run Nala UI tests on multiple browsers/viewports
     - Chrome, Firefox, WebKit browsers
     - Desktop and Mobile viewports

Resolves: [MWPW-157346](https://jira.corp.adobe.com/browse/MWPW-157346)

** Nala Run Command**
image

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://nala--milo--skumar09.hlx.page/?martech=off

Note: 
- This is an initial PR for moving Nala UI tests to Milo Repo, 
- Upon merge of this PR, will start moving other Milo blocks/features UI tests from the Nala repository
- The object of this PR is also to make sure, the defined folder structure, and update to package.json, are not impacting existing features and functionalities.

### CODE REVIEW

**Review by narcis-radu — APPROVED** [review]

**Review by JackySun9 — APPROVED** [review]

**Review by Dli3 — APPROVED** [review]

File: nala/utils/base-reporter.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        console.log('----Failed to publish result to slack channel----');
+      }
+    }
+  }
+
+  printResultSummary() {
+    const totalTests = this.results.length;
+    const passPercentage = ((this.passedTests / totalTests) * 100).toFixed(2);
+    const failPercentage = ((this.failedTests / totalTests) * 100).toFixed(2);
+    const miloLibs = process.env.MILO_LIBS || '';
```
Comment (mokimo) [inline nala/utils/base-reporter.js:97]: Same with the other process variables, have they been added to milo as well?

File: playwright.config.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
[the remark this answers is not in the document]
Comment (skumar09) [reply playwright.config.js:1]: @mokimo: Once we move all Milo Nala tests, and add/update the GitHub action flow to run Nala tests from the Milo repo, i will move this under the Nala folder or change the file name to `nala.config.js`

**Comment by skumar09** [comment]
> Cool a couple of points and questions:
> 
> * Should we update the workflow to always run on the milo repo?
  -> Yeah, we will enable this once we move all Milo tests from Nala repo to Milo repo.
> * I guess the consumer tests will also be moved soon?
  -> Consumer test to consumer repo movement will take some time, 
> I'd recommend adding the nala folder https://github.com/adobecom/milo/blob/stage/.github/workflows/label-zero-impact.js#L2-L17 to this workflow. Zero-impact PRs are generally mergeable faster and with less hassle.
-> Good idea, yeah we should make zero-impact,

File: nala/utils/base-reporter.js (hunk 1 above)
Comment (skumar09) [reply nala/utils/base-reporter.js:97]: > Same with the other process variables, have they been added to milo as well?
-> This process variable is set/updated as part of `run-nala-tests-on-milolibs` GitHub action flow,

**Pull request merged**